File: 003FizzBuzz/Python/fizzbuzz.py


BeginValue = 1
EndValue = 101



def fizzbuzz():
    for num in range(BeginValue, EndValue):
        if num%15 == 0:
            print ('FizzBuzz')
        elif num%3 == 0:
            print ('Fizz')
        elif num%5 == 0:
            print('Buzz')
        else:
            print(num)


if __name__ == '__main__':
    fizzbuzz()

File: 003FizzBuzz/Haskell/fizzbuzz.hs

import Data.Maybe

fizzbuzz :: Int -> IO ()
fizzbuzz 101 = putStrLn ""
fizzbuzz num = do   if num`mod`15 == 0 then putStrLn "FizzBuzz"
                    else if num`mod`3 == 0 then putStrLn "Fizz"
                    else if num`mod`5 == 0 then putStrLn "Buzz"
                    else print num
                    fizzbuzz (num+1)
                    
main = fizzbuzz 1

